<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Install Conda packages from System Requirements section of package Description file • CondaSysReqs</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Install Conda packages from System Requirements section of package Description file">
<meta property="og:description" content="Install Conda packages from System Requirements section of package Description file.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">CondaSysReqs</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/CondaSysReqs.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="introduction" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1></div>
<p>The CondaSysReqs package offers a simple toolset to install and manage Conda environments from R using the the <strong>install_CondaTools</strong> and <strong>install_CondaSysReqs</strong> functions.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Use the <code>BiocManager</code> package to download and install the package from our Github repository:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</span>
<span id="cb1-2"><a href="#cb1-2"></a>    <span class="kw">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-3"><a href="#cb1-3"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">"https://github.com/RockefellerUniversity/CondaSysReqs"</span>)</span></code></pre></div>
<p>Once installed, load it into your R session:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(CondaSysReqs)</span></code></pre></div>
</div>
<div id="simple-install-conda-environment-from-r-console-using-install_condatools" class="section level1">
<h1 class="hasAnchor">
<a href="#simple-install-conda-environment-from-r-console-using-install_condatools" class="anchor"></a>Simple install Conda Environment from R console using <strong>install_CondaTools</strong>.</h1>
<p>The <strong>install_CondaTools()</strong> function allows the user to specify required Conda software and the desired environment to install into.</p>
<p>Miniconda is installed as part of the process (by default into the r-reticulate’s default Conda location).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">install_CondaTools</span>(<span class="st">"salmon"</span>,<span class="st">"myCondaToolSet"</span>)</span>
<span id="cb3-2"><a href="#cb3-2"></a></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co">## $pathToConda</span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="co">## [1] "/Users/thomascarroll/Library/r-miniconda/bin/conda"</span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co">## </span></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="co">## $environment</span></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co">## [1] "myCondaToolSet"</span></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co">## </span></span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="co">## $pathToEnvBin</span></span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="co">## [1] "/Users/thomascarroll/Library/r-miniconda/envs/myCondaToolSet/bin"</span></span></code></pre></div>
<p>We can add additional tools to our Conda environment by specifying <em>updateEnv = TRUE</em>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>pathToConda &lt;-<span class="st"> </span><span class="kw">install_CondaTools</span>(<span class="st">"macs2"</span>,<span class="st">"myCondaToolSet"</span>,<span class="dt">updateEnv =</span> <span class="ot">TRUE</span>)</span>
<span id="cb4-2"><a href="#cb4-2"></a>pathToConda</span>
<span id="cb4-3"><a href="#cb4-3"></a></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co">## $pathToConda</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="co">## [1] "/Users/thomascarroll/Library/r-miniconda/bin/conda"</span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co">## </span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="co">## $environment</span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="co">## [1] "myCondaToolSet"</span></span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="co">## </span></span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="co">## $pathToEnvBin</span></span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="co">## [1] "/Users/thomascarroll/Library/r-miniconda/envs/myCondaToolSet/bin"</span></span></code></pre></div>
<p>Although we will not activate the environment, many tools can be used straight from the Conda environment’s bin directory.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>pathToMacs &lt;-<span class="st"> </span><span class="kw">file.path</span>(pathToConda<span class="op">$</span>pathToEnvBin,<span class="st">"macs2"</span>)</span>
<span id="cb5-2"><a href="#cb5-2"></a>pathToMacs</span>
<span id="cb5-3"><a href="#cb5-3"></a></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="co">## [1] "/Users/thomascarroll/Library/r-miniconda/envs/myCondaToolSet/bin/macs2"</span></span>
<span id="cb5-5"><a href="#cb5-5"></a></span>
<span id="cb5-6"><a href="#cb5-6"></a>Macs_help &lt;-<span class="st"> </span><span class="kw">system</span>(<span class="kw">paste</span>(pathToMacs,<span class="st">"-h"</span>),<span class="dt">intern =</span> <span class="ot">TRUE</span>)</span>
<span id="cb5-7"><a href="#cb5-7"></a>Macs_help</span>
<span id="cb5-8"><a href="#cb5-8"></a></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co">##  [1] "usage: macs2 [-h] [--version]"                                                                                                 </span></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="co">##  [2] "             {callpeak,bdgpeakcall,bdgbroadcall,bdgcmp,bdgopt,cmbreps,bdgdiff,filterdup,predictd,pileup,randsample,refinepeak}"</span></span>
<span id="cb5-11"><a href="#cb5-11"></a><span class="co">##  [3] "             ..."                                                                                                              </span></span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="co">##  [4] ""                                                                                                                              </span></span>
<span id="cb5-13"><a href="#cb5-13"></a><span class="co">##  [5] "macs2 -- Model-based Analysis for ChIP-Sequencing"                                                                             </span></span>
<span id="cb5-14"><a href="#cb5-14"></a><span class="co">##  [6] ""                                                                                                                              </span></span>
<span id="cb5-15"><a href="#cb5-15"></a><span class="co">##  [7] "positional arguments:"                                                                                                         </span></span>
<span id="cb5-16"><a href="#cb5-16"></a><span class="co">##  [8] "  {callpeak,bdgpeakcall,bdgbroadcall,bdgcmp,bdgopt,cmbreps,bdgdiff,filterdup,predictd,pileup,randsample,refinepeak}"           </span></span>
<span id="cb5-17"><a href="#cb5-17"></a><span class="co">##  [9] "    callpeak            Main MACS2 Function: Call peaks from alignment"                                                        </span></span>
<span id="cb5-18"><a href="#cb5-18"></a><span class="co">## [10] "                        results."                                                                                              </span></span>
<span id="cb5-19"><a href="#cb5-19"></a><span class="co">## [11] "    bdgpeakcall         Call peaks from bedGraph output. Note: All regions on"                                                 </span></span>
<span id="cb5-20"><a href="#cb5-20"></a><span class="co">## [12] "                        the same chromosome in the bedGraph file should be"                                                    </span></span>
<span id="cb5-21"><a href="#cb5-21"></a><span class="co">## [13] "                        continuous so only bedGraph files from MACS2 are"                                                      </span></span>
<span id="cb5-22"><a href="#cb5-22"></a><span class="co">## [14] "                        accpetable."                                                                                           </span></span>
<span id="cb5-23"><a href="#cb5-23"></a><span class="co">## [15] "    bdgbroadcall        Call broad peaks from bedGraph output. Note: All"                                                      </span></span>
<span id="cb5-24"><a href="#cb5-24"></a><span class="co">## [16] "                        regions on the same chromosome in the bedGraph file"                                                   </span></span>
<span id="cb5-25"><a href="#cb5-25"></a><span class="co">## [17] "                        should be continuous so only bedGraph files from MACS2"                                                </span></span>
<span id="cb5-26"><a href="#cb5-26"></a><span class="co">## [18] "                        are accpetable."                                                                                       </span></span>
<span id="cb5-27"><a href="#cb5-27"></a><span class="co">## [19] "    bdgcmp              Deduct noise by comparing two signal tracks in"                                                        </span></span>
<span id="cb5-28"><a href="#cb5-28"></a><span class="co">## [20] "                        bedGraph. Note: All regions on the same chromosome in"                                                 </span></span>
<span id="cb5-29"><a href="#cb5-29"></a><span class="co">## [21] "                        the bedGraph file should be continuous so only"                                                        </span></span>
<span id="cb5-30"><a href="#cb5-30"></a><span class="co">## [22] "                        bedGraph files from MACS2 are accpetable."                                                             </span></span>
<span id="cb5-31"><a href="#cb5-31"></a><span class="co">## [23] "    bdgopt              Operations on score column of bedGraph file. Note: All"                                                </span></span>
<span id="cb5-32"><a href="#cb5-32"></a><span class="co">## [24] "                        regions on the same chromosome in the bedGraph file"                                                   </span></span>
<span id="cb5-33"><a href="#cb5-33"></a><span class="co">## [25] "                        should be continuous so only bedGraph files from MACS2"                                                </span></span>
<span id="cb5-34"><a href="#cb5-34"></a><span class="co">## [26] "                        are accpetable."                                                                                       </span></span>
<span id="cb5-35"><a href="#cb5-35"></a><span class="co">## [27] "    cmbreps             Combine BEDGraphs of scores from replicates. Note: All"                                                </span></span>
<span id="cb5-36"><a href="#cb5-36"></a><span class="co">## [28] "                        regions on the same chromosome in the bedGraph file"                                                   </span></span>
<span id="cb5-37"><a href="#cb5-37"></a><span class="co">## [29] "                        should be continuous so only bedGraph files from MACS2"                                                </span></span>
<span id="cb5-38"><a href="#cb5-38"></a><span class="co">## [30] "                        are accpetable."                                                                                       </span></span>
<span id="cb5-39"><a href="#cb5-39"></a><span class="co">## [31] "    bdgdiff             Differential peak detection based on paired four"                                                      </span></span>
<span id="cb5-40"><a href="#cb5-40"></a><span class="co">## [32] "                        bedgraph files. Note: All regions on the same"                                                         </span></span>
<span id="cb5-41"><a href="#cb5-41"></a><span class="co">## [33] "                        chromosome in the bedGraph file should be continuous"                                                  </span></span>
<span id="cb5-42"><a href="#cb5-42"></a><span class="co">## [34] "                        so only bedGraph files from MACS2 are accpetable."                                                     </span></span>
<span id="cb5-43"><a href="#cb5-43"></a><span class="co">## [35] "    filterdup           Remove duplicate reads at the same position, then save"                                                </span></span>
<span id="cb5-44"><a href="#cb5-44"></a><span class="co">## [36] "                        the rest alignments to BED or BEDPE file. If you use '"                                                </span></span>
<span id="cb5-45"><a href="#cb5-45"></a><span class="co">## [37] "                        --keep-dup all option', this script can be utilized to"                                                </span></span>
<span id="cb5-46"><a href="#cb5-46"></a><span class="co">## [38] "                        convert any acceptable format into BED or BEDPE"                                                       </span></span>
<span id="cb5-47"><a href="#cb5-47"></a><span class="co">## [39] "                        format."                                                                                               </span></span>
<span id="cb5-48"><a href="#cb5-48"></a><span class="co">## [40] "    predictd            Predict d or fragment size from alignment results."                                                    </span></span>
<span id="cb5-49"><a href="#cb5-49"></a><span class="co">## [41] "                        *Will NOT filter duplicates*"                                                                          </span></span>
<span id="cb5-50"><a href="#cb5-50"></a><span class="co">## [42] "    pileup              Pileup aligned reads with a given extension size"                                                      </span></span>
<span id="cb5-51"><a href="#cb5-51"></a><span class="co">## [43] "                        (fragment size or d in MACS language). Note there will"                                                </span></span>
<span id="cb5-52"><a href="#cb5-52"></a><span class="co">## [44] "                        be no step for duplicate reads filtering or sequencing"                                                </span></span>
<span id="cb5-53"><a href="#cb5-53"></a><span class="co">## [45] "                        depth scaling, so you may need to do certain pre/post-"                                                </span></span>
<span id="cb5-54"><a href="#cb5-54"></a><span class="co">## [46] "                        processing."                                                                                           </span></span>
<span id="cb5-55"><a href="#cb5-55"></a><span class="co">## [47] "    randsample          Randomly sample number/percentage of total reads."                                                     </span></span>
<span id="cb5-56"><a href="#cb5-56"></a><span class="co">## [48] "    refinepeak          (Experimental) Take raw reads alignment, refine peak"                                                  </span></span>
<span id="cb5-57"><a href="#cb5-57"></a><span class="co">## [49] "                        summits and give scores measuring balance of"                                                          </span></span>
<span id="cb5-58"><a href="#cb5-58"></a><span class="co">## [50] "                        waston/crick tags. Inspired by SPP."                                                                   </span></span>
<span id="cb5-59"><a href="#cb5-59"></a><span class="co">## [51] ""                                                                                                                              </span></span>
<span id="cb5-60"><a href="#cb5-60"></a><span class="co">## [52] "optional arguments:"                                                                                                           </span></span>
<span id="cb5-61"><a href="#cb5-61"></a><span class="co">## [53] "  -h, --help            show this help message and exit"                                                                       </span></span>
<span id="cb5-62"><a href="#cb5-62"></a><span class="co">## [54] "  --version             show program's version number and exit"                                                                </span></span>
<span id="cb5-63"><a href="#cb5-63"></a><span class="co">## [55] ""                                                                                                                              </span></span>
<span id="cb5-64"><a href="#cb5-64"></a><span class="co">## [56] "For command line options of each command, type: macs2 COMMAND -h"</span></span>
<span id="cb5-65"><a href="#cb5-65"></a></span>
<span id="cb5-66"><a href="#cb5-66"></a>pathToSalmon &lt;-<span class="st"> </span><span class="kw">file.path</span>(pathToConda<span class="op">$</span>pathToEnvBin,<span class="st">"salmon"</span>)</span>
<span id="cb5-67"><a href="#cb5-67"></a>pathToSalmon</span>
<span id="cb5-68"><a href="#cb5-68"></a></span>
<span id="cb5-69"><a href="#cb5-69"></a><span class="co">## [1] "/Users/thomascarroll/Library/r-miniconda/envs/myCondaToolSet/bin/salmon"</span></span>
<span id="cb5-70"><a href="#cb5-70"></a></span>
<span id="cb5-71"><a href="#cb5-71"></a>Salmon_help &lt;-<span class="st"> </span><span class="kw">system</span>(<span class="kw">paste</span>(pathToSalmon,<span class="st">"-h"</span>),<span class="dt">intern =</span> <span class="ot">TRUE</span>)</span>
<span id="cb5-72"><a href="#cb5-72"></a>Salmon_help</span>
<span id="cb5-73"><a href="#cb5-73"></a></span>
<span id="cb5-74"><a href="#cb5-74"></a><span class="co">##  [1] "salmon v1.3.0"                                                      </span></span>
<span id="cb5-75"><a href="#cb5-75"></a><span class="co">##  [2] ""                                                                   </span></span>
<span id="cb5-76"><a href="#cb5-76"></a><span class="co">##  [3] "Usage:  salmon -h|--help or "                                       </span></span>
<span id="cb5-77"><a href="#cb5-77"></a><span class="co">##  [4] "        salmon -v|--version or "                                    </span></span>
<span id="cb5-78"><a href="#cb5-78"></a><span class="co">##  [5] "        salmon -c|--cite or "                                       </span></span>
<span id="cb5-79"><a href="#cb5-79"></a><span class="co">##  [6] "        salmon [--no-version-check] &lt;COMMAND&gt; [-h | options]"       </span></span>
<span id="cb5-80"><a href="#cb5-80"></a><span class="co">##  [7] ""                                                                   </span></span>
<span id="cb5-81"><a href="#cb5-81"></a><span class="co">##  [8] "Commands:"                                                          </span></span>
<span id="cb5-82"><a href="#cb5-82"></a><span class="co">##  [9] "     index      : create a salmon index"                            </span></span>
<span id="cb5-83"><a href="#cb5-83"></a><span class="co">## [10] "     quant      : quantify a sample"                                </span></span>
<span id="cb5-84"><a href="#cb5-84"></a><span class="co">## [11] "     alevin     : single cell analysis"                             </span></span>
<span id="cb5-85"><a href="#cb5-85"></a><span class="co">## [12] "     swim       : perform super-secret operation"                   </span></span>
<span id="cb5-86"><a href="#cb5-86"></a><span class="co">## [13] "     quantmerge : merge multiple quantifications into a single file"</span></span></code></pre></div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Thomas Carroll <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/??" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Matt Paul <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3020-7729" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Doug Barrows <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/??" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Thomas Carroll, Matt Paul, Doug Barrows.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.0.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
