<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Installing and managing Conda environments with the CondaSysReqs package • CondaSysReqs</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Installing and managing Conda environments with the CondaSysReqs package">
<meta property="og:description" content="CondaSysReqs">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CondaSysReqs</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CondaSysReqs.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="CondaSysReqs_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Installing and managing Conda environments with the CondaSysReqs package</h1>
                        <h4 class="author">Tom Carroll, Doug Barrows and Matt Paul</h4>
            
            <h4 class="date">09/10/2020</h4>
      
      
      <div class="hidden name"><code>CondaSysReqs.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The CondaSysReqs package offers a simple toolset to install and manage Conda environments from R using the the <strong>install_CondaTools</strong> and <strong>install_CondaSysReqs</strong> functions.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Use the <code>BiocManager</code> package to download and install the package from our Github repository:</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="co">if</span> (<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, quietly = <span class="fl">TRUE</span>))
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
<span class="kw">BiocManager</span>::<span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"https://github.com/RockefellerUniversity/CondaSysReqs"</span>)
</pre></div>
<p>Once installed, load it into your R session:</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">CondaSysReqs</span>)
</pre></div>
</div>
<div id="simple-install-conda-environment-from-r-console-using-install_condatools-" class="section level1">
<h1 class="hasAnchor">
<a href="#simple-install-conda-environment-from-r-console-using-install_condatools-" class="anchor"></a>Simple install Conda Environment from R console using <strong>install_CondaTools</strong>.</h1>
<p>The <strong>install_CondaTools()</strong> function allows the user to specify required Conda software and the desired environment to install into.</p>
<p>Miniconda is installed as part of the process (by default into the r-reticulate’s default Conda location).</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="fu"><a href="../reference/install_CondaTools.html">install_CondaTools</a></span>(<span class="st">"salmon"</span>,<span class="st">"myCondaToolSet"</span>)
</pre></div>
<pre><code>## $pathToConda
## [1] "/Users/runner/Library/r-miniconda/bin/conda"
## 
## $environment
## [1] "myCondaToolSet"
## 
## $pathToEnvBin
## [1] "/Users/runner/Library/r-miniconda/envs/myCondaToolSet/bin"</code></pre>
<p>We can add additional tools to our Conda environment by specifying <em>updateEnv = TRUE</em>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="kw">pathToConda</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/install_CondaTools.html">install_CondaTools</a></span>(<span class="st">"macs2"</span>,<span class="st">"myCondaToolSet"</span>,updateEnv = <span class="fl">TRUE</span>)
<span class="kw">pathToConda</span>
</pre></div>
<pre><code>## $pathToConda
## [1] "/Users/runner/Library/r-miniconda/bin/conda"
## 
## $environment
## [1] "myCondaToolSet"
## 
## $pathToEnvBin
## [1] "/Users/runner/Library/r-miniconda/envs/myCondaToolSet/bin"</code></pre>
<p>Although we will not activate the environment, many tools can be used straight from the Conda environment’s bin directory.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="kw">pathToMacs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">pathToConda</span><span class="op">$</span><span class="kw">pathToEnvBin</span>,<span class="st">"macs2"</span>)
<span class="kw">pathToMacs</span>
</pre></div>
<pre><code>## [1] "/Users/runner/Library/r-miniconda/envs/myCondaToolSet/bin/macs2"</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="kw">Macs_help</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.html">system</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="kw">pathToMacs</span>,<span class="st">"-h"</span>),intern = <span class="fl">TRUE</span>)
<span class="kw">Macs_help</span>
</pre></div>
<pre><code>##  [1] "usage: macs2 [-h] [--version]"                                                                                                 
##  [2] "             {callpeak,bdgpeakcall,bdgbroadcall,bdgcmp,bdgopt,cmbreps,bdgdiff,filterdup,predictd,pileup,randsample,refinepeak}"
##  [3] "             ..."                                                                                                              
##  [4] ""                                                                                                                              
##  [5] "macs2 -- Model-based Analysis for ChIP-Sequencing"                                                                             
##  [6] ""                                                                                                                              
##  [7] "positional arguments:"                                                                                                         
##  [8] "  {callpeak,bdgpeakcall,bdgbroadcall,bdgcmp,bdgopt,cmbreps,bdgdiff,filterdup,predictd,pileup,randsample,refinepeak}"           
##  [9] "    callpeak            Main MACS2 Function: Call peaks from alignment"                                                        
## [10] "                        results."                                                                                              
## [11] "    bdgpeakcall         Call peaks from bedGraph output. Note: All regions on"                                                 
## [12] "                        the same chromosome in the bedGraph file should be"                                                    
## [13] "                        continuous so only bedGraph files from MACS2 are"                                                      
## [14] "                        accpetable."                                                                                           
## [15] "    bdgbroadcall        Call broad peaks from bedGraph output. Note: All"                                                      
## [16] "                        regions on the same chromosome in the bedGraph file"                                                   
## [17] "                        should be continuous so only bedGraph files from MACS2"                                                
## [18] "                        are accpetable."                                                                                       
## [19] "    bdgcmp              Deduct noise by comparing two signal tracks in"                                                        
## [20] "                        bedGraph. Note: All regions on the same chromosome in"                                                 
## [21] "                        the bedGraph file should be continuous so only"                                                        
## [22] "                        bedGraph files from MACS2 are accpetable."                                                             
## [23] "    bdgopt              Operations on score column of bedGraph file. Note: All"                                                
## [24] "                        regions on the same chromosome in the bedGraph file"                                                   
## [25] "                        should be continuous so only bedGraph files from MACS2"                                                
## [26] "                        are accpetable."                                                                                       
## [27] "    cmbreps             Combine BEDGraphs of scores from replicates. Note: All"                                                
## [28] "                        regions on the same chromosome in the bedGraph file"                                                   
## [29] "                        should be continuous so only bedGraph files from MACS2"                                                
## [30] "                        are accpetable."                                                                                       
## [31] "    bdgdiff             Differential peak detection based on paired four"                                                      
## [32] "                        bedgraph files. Note: All regions on the same"                                                         
## [33] "                        chromosome in the bedGraph file should be continuous"                                                  
## [34] "                        so only bedGraph files from MACS2 are accpetable."                                                     
## [35] "    filterdup           Remove duplicate reads at the same position, then save"                                                
## [36] "                        the rest alignments to BED or BEDPE file. If you use '"                                                
## [37] "                        --keep-dup all option', this script can be utilized to"                                                
## [38] "                        convert any acceptable format into BED or BEDPE"                                                       
## [39] "                        format."                                                                                               
## [40] "    predictd            Predict d or fragment size from alignment results."                                                    
## [41] "                        *Will NOT filter duplicates*"                                                                          
## [42] "    pileup              Pileup aligned reads with a given extension size"                                                      
## [43] "                        (fragment size or d in MACS language). Note there will"                                                
## [44] "                        be no step for duplicate reads filtering or sequencing"                                                
## [45] "                        depth scaling, so you may need to do certain pre/post-"                                                
## [46] "                        processing."                                                                                           
## [47] "    randsample          Randomly sample number/percentage of total reads."                                                     
## [48] "    refinepeak          (Experimental) Take raw reads alignment, refine peak"                                                  
## [49] "                        summits and give scores measuring balance of"                                                          
## [50] "                        waston/crick tags. Inspired by SPP."                                                                   
## [51] ""                                                                                                                              
## [52] "optional arguments:"                                                                                                           
## [53] "  -h, --help            show this help message and exit"                                                                       
## [54] "  --version             show program's version number and exit"                                                                
## [55] ""                                                                                                                              
## [56] "For command line options of each command, type: macs2 COMMAND -h"</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="kw">pathToSalmon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">pathToConda</span><span class="op">$</span><span class="kw">pathToEnvBin</span>,<span class="st">"salmon"</span>)
<span class="kw">pathToSalmon</span>
</pre></div>
<pre><code>## [1] "/Users/runner/Library/r-miniconda/envs/myCondaToolSet/bin/salmon"</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="kw">Salmon_help</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.html">system</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="kw">pathToSalmon</span>,<span class="st">"-h"</span>),intern = <span class="fl">TRUE</span>)
<span class="kw">Salmon_help</span>
</pre></div>
<pre><code>##  [1] "salmon v1.3.0"                                                      
##  [2] ""                                                                   
##  [3] "Usage:  salmon -h|--help or "                                       
##  [4] "        salmon -v|--version or "                                    
##  [5] "        salmon -c|--cite or "                                       
##  [6] "        salmon [--no-version-check] &lt;COMMAND&gt; [-h | options]"       
##  [7] ""                                                                   
##  [8] "Commands:"                                                          
##  [9] "     index      : create a salmon index"                            
## [10] "     quant      : quantify a sample"                                
## [11] "     alevin     : single cell analysis"                             
## [12] "     swim       : perform super-secret operation"                   
## [13] "     quantmerge : merge multiple quantifications into a single file"</code></pre>
</div>
<div id="acknowledgements" class="section level1">
<h1 class="hasAnchor">
<a href="#acknowledgements" class="anchor"></a>Acknowledgements</h1>
<p>Thank you to Ji-Dung Luo and Wei Wang for testing/vignette review/critical feedback and Ziwei Liang for their support.</p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
</pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] CondaSysReqs_0.9.8 reticulate_1.16    BiocStyle_2.16.0  
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5          cpp11_0.2.1         knitr_1.29          magrittr_1.5       
##  [5] lattice_0.20-41     rjson_0.2.20        R6_2.4.1            ragg_0.3.1         
##  [9] rlang_0.4.7         stringr_1.4.0       tools_4.0.2         grid_4.0.2         
## [13] xfun_0.17           htmltools_0.5.0     systemfonts_0.3.1   yaml_2.2.1         
## [17] assertthat_0.2.1    rprojroot_1.3-2     digest_0.6.25       pkgdown_1.6.0      
## [21] crayon_1.3.4        bookdown_0.20       Matrix_1.2-18       BiocManager_1.30.10
## [25] fs_1.5.0            memoise_1.1.0       evaluate_0.14       rmarkdown_2.3      
## [29] stringi_1.5.3       compiler_4.0.2      desc_1.2.0          backports_1.1.9    
## [33] jsonlite_1.7.1</code></pre>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Thomas Carroll, Matt Paul, Doug Barrows.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.0.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
