<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Herper Guide • Herper</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Herper Guide">
<meta property="og:description" content="Herper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Herper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Herper.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QuickStart.html">Herper Quick Start Guide</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RockefellerUniversity/Herper/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Herper_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Herper Guide</h1>
                        <h4 class="author">Matt Paul - <a href="mailto:mpaul@rockefeller.edu" class="email">mpaul@rockefeller.edu</a>
</h4>
            
            <h4 class="date">02/22/2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RockefellerUniversity/Herper/blob/master/vignettes/Herper.Rmd"><code>vignettes/Herper.Rmd</code></a></small>
      <div class="hidden name"><code>Herper.Rmd</code></div>

    </div>

    
    
<p><br></p>
<hr>
<div id="what-is-herper" class="section level2">
<h2 class="hasAnchor">
<a href="#what-is-herper" class="anchor"></a>What is Herper?</h2>
<p>The Herper package is a simple toolset to install and manage Conda packages and environments from within the R console.</p>
<p>Unfortunately many tools for data analysis are not available in R, but are present in public repositories like conda. With Herper users can install, manage, record and run conda tools from the comfort of their R session.</p>
<p>Furthermore, many R packages require the use of these external dependencies. Again these dependencies can be installed and managed with the Conda package repository. For example 169 Bioconductor packages have external dependencies listed in their System Requirements field (often with these packages having several requirements) [03 September, 2020].</p>
<p><br></p>
<p><img src="../../../_temp/Library/Herper/extdata/pkg_deps_bar_mask-1.png" width="1000px" style="display: block; margin: auto;"></p>
<p>Herper provides an ad-hoc approach to handling external system requirements for R packages. For people developing packages with python conda dependencies we recommend using <a href="https://bioconductor.org/packages/release/bioc/html/basilisk.html">basilisk</a> to internally support these system requirements pre-hoc.</p>
<p>The Herper package was developed by <a href="https://github.com/matthew-paul-2006">Matt Paul</a>, <a href="https://github.com/dougbarrows">Doug Barrows</a> and <a href="https://github.com/ThomasCarroll">Thomas Carroll</a> at the <a href="https://rockefelleruniversity.github.io">Rockefeller University Bioinformatics Resources Center</a> with contributions from <a href="https://github.com/kathrynrozengagnon">Kathryn Rozen-Gagnon</a>.</p>
<p><br></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Use the <code>BiocManager</code> package to download and install the package from our Github repository:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
  <span class="op">}</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"Herper"</span><span class="op">)</span></code></pre></div>
<p><br> Once installed, load it into your R session:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RockefellerUniversity/Herper">Herper</a></span><span class="op">)</span></code></pre></div>
<p><br></p>
</div>
<div id="simple-install-of-conda-packages-from-r-console-using-install_condatools-" class="section level2">
<h2 class="hasAnchor">
<a href="#simple-install-of-conda-packages-from-r-console-using-install_condatools-" class="anchor"></a>Simple install of Conda packages from R console using <strong>install_CondaTools</strong>.</h2>
<p>The <strong>install_CondaTools()</strong> function allows the user to specify required Conda software and the desired environment to install into.</p>
<p>Miniconda is installed as part of the process (by default into the r-reticulate’s default Conda location - /Users/runner/Library/r-miniconda) and the user’s requested conda environment built within the same directory (by default /Users/runner/Library/r-miniconda/envs/USERS_ENVIRONMENT_HERE).</p>
<p>If you already have Miniconda installed or you would like to install to a custom location, you can specify the path with the <em>pathToMiniConda</em> parameter. In this example we are installing in a temporary directory, but most likely you will want to install/use a stable version of Miniconda.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">myMiniconda</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu">tempdir2</span><span class="op">(</span><span class="op">)</span>, <span class="st">"Test"</span><span class="op">)</span>
<span class="va">myMiniconda</span></code></pre></div>
<pre><code>## [1] "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpE46kLy/rr/Test"</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/install_CondaTools.html">install_CondaTools</a></span><span class="op">(</span><span class="st">"samtools"</span>, <span class="st">"herper"</span>, pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span></code></pre></div>
<pre><code>## $pathToConda
## [1] "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpE46kLy/rr/Test/bin/conda"
## 
## $environment
## [1] "herper"
## 
## $pathToEnvBin
## [1] "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpE46kLy/rr/Test/envs/herper/bin"</code></pre>
<p><br> We can add additional tools to our Conda environment by specifying <em>updateEnv = TRUE</em>. A vector of tools can be used to install several at once.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pathToConda</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/install_CondaTools.html">install_CondaTools</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"salmon"</span>, <span class="st">"kallisto"</span><span class="op">)</span>, <span class="st">"herper"</span>, updateEnv <span class="op">=</span> <span class="cn">TRUE</span>, pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span>

<span class="va">pathToConda</span></code></pre></div>
<pre><code>## $pathToConda
## [1] "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpE46kLy/rr/Test/bin/conda"
## 
## $environment
## [1] "herper"
## 
## $pathToEnvBin
## [1] "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpE46kLy/rr/Test/envs/herper/bin"</code></pre>
<p><br> Specific package versions can be installed using conda formatted inputs into the <em>tools</em> argument i.e. “salmon==1.3”, “salmon&gt;=1.3” or “salmon&lt;=1.3”. This can also be used to specifically upgrade or downgrade existing tools in the chosen environment.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pathToConda</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/install_CondaTools.html">install_CondaTools</a></span><span class="op">(</span><span class="st">"salmon&lt;=1.3"</span>, <span class="st">"herper"</span>, updateEnv <span class="op">=</span> <span class="cn">TRUE</span>, pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span></code></pre></div>
<p><br></p>
</div>
<div id="install-r-package-dependencies-with-install_condasysreqs-" class="section level2">
<h2 class="hasAnchor">
<a href="#install-r-package-dependencies-with-install_condasysreqs-" class="anchor"></a>Install R package dependencies with <strong>install_CondaSysReqs</strong>.</h2>
<p>The <strong>install_CondaSysReqs</strong> checks the System Requirements for the specified R package, and uses Conda to install this software. Here we will use a test package contained within Herper. This test package has two System Requirements:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">testPkg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/HerperTestPkg"</span>, package <span class="op">=</span> <span class="st">"Herper"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="va">testPkg</span>, type <span class="op">=</span> <span class="st">"source"</span>, repos <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
<span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageDescription</a></span><span class="op">(</span><span class="st">"HerperTestPkg"</span>, fields <span class="op">=</span> <span class="st">"SystemRequirements"</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "samtools==1.10, rmats&gt;=v4.1.0"</code></pre>
<p>The user can simply supply the name of an installed R package, and <strong>install_CondaSysReqs</strong> will install the System Requirements through conda.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/install_CondaSysReqs.html">install_CondaSysReqs</a></span><span class="op">(</span><span class="st">"HerperTestPkg"</span>, pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span></code></pre></div>
<pre><code>## $pathToConda
## [1] "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpE46kLy/rr/Test/bin/conda"
## 
## $environment
## [1] "HerperTestPkg_0.1.0"
## 
## $pathToEnvBin
## [1] "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpE46kLy/rr/Test/envs/HerperTestPkg_0.1.0/bin"</code></pre>
<p>By default these packages are installed in a new environment, which has the name name of the R package and its version number. Users can control the environment name using the <em>env</em> parameter. As with <strong>install_CondaTools()</strong>, user can control which version of Miniconda with the parameter <em>pathToMiniConda</em>, and whether they want to amend an existing environment with the parameter <em>updateEnv</em>.</p>
<p><em>Note: <strong>install_CondaSysReqs</strong> can handle standard System Requirement formats, but will not work if the package has free form text. In this case just use <strong>install_CondaTools</strong></em></p>
<p><br></p>
</div>
<div id="using-external-software-with-the-with_condaenv-and-local_condaenv-functions-" class="section level2">
<h2 class="hasAnchor">
<a href="#using-external-software-with-the-with_condaenv-and-local_condaenv-functions-" class="anchor"></a>Using external software with the <strong>with_CondaEnv</strong> and <strong>local_CondaEnv</strong> functions.</h2>
<p>Once installed within a conda environment, many external software can be executed directly from the conda environment’s bin directory without having to perform any additional actions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pathToSamtools</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">pathToConda</span><span class="op">$</span><span class="va">pathToEnvBin</span>,<span class="st">"samtools"</span><span class="op">)</span>
<span class="va">Res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system2.html">system2</a></span><span class="op">(</span>command<span class="op">=</span><span class="va">pathToSamtools</span>, args <span class="op">=</span> <span class="st">"help"</span>,stdout <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">Res</span></code></pre></div>
<pre><code>## 
## Program: samtools (Tools for alignments in the SAM format)
## Version: 1.11 (using htslib 1.11)
## 
## Usage:   samtools &lt;command&gt; [options]
## 
## Commands:
##   -- Indexing
##      dict           create a sequence dictionary file
##      faidx          index/extract FASTA
##      fqidx          index/extract FASTQ
##      index          index alignment
## 
##   -- Editing
##      calmd          recalculate MD/NM tags and '=' bases
##      fixmate        fix mate information
##      reheader       replace BAM header
##      targetcut      cut fosmid regions (for fosmid pool only)
##      addreplacerg   adds or replaces RG tags
##      markdup        mark duplicates
##      ampliconclip   clip oligos from the end of reads
## 
##   -- File operations
##      collate        shuffle and group alignments by name
##      cat            concatenate BAMs
##      merge          merge sorted alignments
##      mpileup        multi-way pileup
##      sort           sort alignment file
##      split          splits a file by read group
##      quickcheck     quickly check if SAM/BAM/CRAM file appears intact
##      fastq          converts a BAM to a FASTQ
##      fasta          converts a BAM to a FASTA
## 
##   -- Statistics
##      bedcov         read depth per BED region
##      coverage       alignment depth and percent coverage
##      depth          compute the depth
##      flagstat       simple stats
##      idxstats       BAM index stats
##      phase          phase heterozygotes
##      stats          generate stats (former bamcheck)
##      ampliconstats  generate amplicon specific stats
## 
##   -- Viewing
##      flags          explain BAM flags
##      tview          text alignment viewer
##      view           SAM&lt;-&gt;BAM&lt;-&gt;CRAM conversion
##      depad          convert padded BAM to unpadded BAM</code></pre>
<p>Some external software however require additional environmental variable to be set in order to execute correctly. An example of this would be <strong>Cytoscape</strong> which requires the java home directory and java library paths to be set prior to its execution.</p>
<p>The Herper package uses the <strong><a href="https://withr.r-lib.org">withr</a></strong> family of functions (<strong>with_CondaEnv()</strong> and <strong>local_CondaEnv()</strong>) to provide methods to <strong>temporarily</strong> alter the system PATH and to add or update any required environmental variables. This is done without formally activating your environment or initializing your conda.</p>
<p>The <strong>with_CondaEnv</strong> allows users to run R code with the required PATH and environmental variables automatically set. The <strong>with_CondaEnv</strong> function simply requires the name of conda environment and the code to be executed within this environment. Additionally we can also the <strong>pathToMiniconda</strong> argument to specify any custom miniconda install location.</p>
<p>The <strong>with_CondaEnv</strong> function will update the PATH we can now run the above samtools command without specifying the full directory path to samtools.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/with_CondaEnv.html">with_CondaEnv</a></span><span class="op">(</span><span class="st">"herper"</span>,
                      <span class="fu"><a href="https://rdrr.io/r/base/system2.html">system2</a></span><span class="op">(</span>command<span class="op">=</span><span class="st">"samtools"</span>,args <span class="op">=</span> <span class="st">"help"</span>,stdout <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                      pathToMiniConda<span class="op">=</span><span class="va">myMiniconda</span><span class="op">)</span>
<span class="va">res</span></code></pre></div>
<pre><code>## 
## Program: samtools (Tools for alignments in the SAM format)
## Version: 1.11 (using htslib 1.11)
## 
## Usage:   samtools &lt;command&gt; [options]
## 
## Commands:
##   -- Indexing
##      dict           create a sequence dictionary file
##      faidx          index/extract FASTA
##      fqidx          index/extract FASTQ
##      index          index alignment
## 
##   -- Editing
##      calmd          recalculate MD/NM tags and '=' bases
##      fixmate        fix mate information
##      reheader       replace BAM header
##      targetcut      cut fosmid regions (for fosmid pool only)
##      addreplacerg   adds or replaces RG tags
##      markdup        mark duplicates
##      ampliconclip   clip oligos from the end of reads
## 
##   -- File operations
##      collate        shuffle and group alignments by name
##      cat            concatenate BAMs
##      merge          merge sorted alignments
##      mpileup        multi-way pileup
##      sort           sort alignment file
##      split          splits a file by read group
##      quickcheck     quickly check if SAM/BAM/CRAM file appears intact
##      fastq          converts a BAM to a FASTQ
##      fasta          converts a BAM to a FASTA
## 
##   -- Statistics
##      bedcov         read depth per BED region
##      coverage       alignment depth and percent coverage
##      depth          compute the depth
##      flagstat       simple stats
##      idxstats       BAM index stats
##      phase          phase heterozygotes
##      stats          generate stats (former bamcheck)
##      ampliconstats  generate amplicon specific stats
## 
##   -- Viewing
##      flags          explain BAM flags
##      tview          text alignment viewer
##      view           SAM&lt;-&gt;BAM&lt;-&gt;CRAM conversion
##      depad          convert padded BAM to unpadded BAM</code></pre>
<p>The <strong>local_CondaEnv</strong> function acts in a similar fashion to the <strong>with_CondaEnv</strong> function and allows the user to temporarily update the required PATH and environmental variable from within a function. The PATH and environmental variables will be modified only until the current function ends.</p>
<p><strong>local_CondaEnv</strong> is best used within a user-created function, allowing access to the Conda environment’s PATH and variables from within the the function itself but resetting all environmental variables once complete.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">samtoolsHelp</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="../reference/with_CondaEnv.html">local_CondaEnv</a></span><span class="op">(</span><span class="st">"herper"</span>, pathToMiniConda<span class="op">=</span><span class="va">myMiniconda</span><span class="op">)</span>
  <span class="va">helpMessage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system2.html">system2</a></span><span class="op">(</span>command<span class="op">=</span><span class="st">"samtools"</span>,args <span class="op">=</span> <span class="st">"help"</span>,stdout <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  <span class="va">helpMessage</span>
<span class="op">}</span>
<span class="fu">samtoolsHelp</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Program: samtools (Tools for alignments in the SAM format)
## Version: 1.11 (using htslib 1.11)
## 
## Usage:   samtools &lt;command&gt; [options]
## 
## Commands:
##   -- Indexing
##      dict           create a sequence dictionary file
##      faidx          index/extract FASTA
##      fqidx          index/extract FASTQ
##      index          index alignment
## 
##   -- Editing
##      calmd          recalculate MD/NM tags and '=' bases
##      fixmate        fix mate information
##      reheader       replace BAM header
##      targetcut      cut fosmid regions (for fosmid pool only)
##      addreplacerg   adds or replaces RG tags
##      markdup        mark duplicates
##      ampliconclip   clip oligos from the end of reads
## 
##   -- File operations
##      collate        shuffle and group alignments by name
##      cat            concatenate BAMs
##      merge          merge sorted alignments
##      mpileup        multi-way pileup
##      sort           sort alignment file
##      split          splits a file by read group
##      quickcheck     quickly check if SAM/BAM/CRAM file appears intact
##      fastq          converts a BAM to a FASTQ
##      fasta          converts a BAM to a FASTA
## 
##   -- Statistics
##      bedcov         read depth per BED region
##      coverage       alignment depth and percent coverage
##      depth          compute the depth
##      flagstat       simple stats
##      idxstats       BAM index stats
##      phase          phase heterozygotes
##      stats          generate stats (former bamcheck)
##      ampliconstats  generate amplicon specific stats
## 
##   -- Viewing
##      flags          explain BAM flags
##      tview          text alignment viewer
##      view           SAM&lt;-&gt;BAM&lt;-&gt;CRAM conversion
##      depad          convert padded BAM to unpadded BAM</code></pre>
<p>To further demonstrate this we will use the first command from the <a href="https://www.bioconductor.org/packages/release/bioc/html/seqCNA.html">seqCNA</a> vignette. This step requires samtools. If this is not installed and available there is an error.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">seqCNA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">seqsumm_HCC1143</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/try.html">try</a></span><span class="op">(</span><span class="va">rco</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/seqCNA/man/readSeqsumm.html">readSeqsumm</a></span><span class="op">(</span>tumour.data <span class="op">=</span> <span class="va">seqsumm_HCC1143</span><span class="op">)</span>, silent <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>Samtools is listed as a System Requirement for seqCNA, so we can first use <strong>install_CondaSysReqs()</strong> to install samtools. In this case we are installing samtools in the environment: seqCNA_env. We can then run the seqCNA command using <strong>with_CondaEnv</strong> specifying that we want to use our environment containing samtools. seqCNA can then find samtools and execute successfully.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/install_CondaSysReqs.html">install_CondaSysReqs</a></span><span class="op">(</span>pkg<span class="op">=</span><span class="st">"seqCNA"</span>,env<span class="op">=</span><span class="st">"seqCNA_env"</span>,pathToMiniConda<span class="op">=</span><span class="va">myMiniconda</span><span class="op">)</span>
<span class="va">rco</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/with_CondaEnv.html">with_CondaEnv</a></span><span class="op">(</span>new<span class="op">=</span><span class="st">"seqCNA_env"</span>,<span class="fu"><a href="https://rdrr.io/pkg/seqCNA/man/readSeqsumm.html">readSeqsumm</a></span><span class="op">(</span>tumour.data<span class="op">=</span><span class="va">seqsumm_HCC1143</span><span class="op">)</span>
 ,pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">rco</span><span class="op">)</span></code></pre></div>
<pre><code>## Basic information:
##   SeqCNAInfo object with 5314 200Kbp-long windows.
##   PEM information is not available.
##   Paired normal is not available.
##   Genome and build unknown (chromosomes chr1 to chr5).
## The profile is not yet normalized and not yet segmented.</code></pre>
<p><br></p>
</div>
<div id="finding-conda-packages-with-conda_search" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-conda-packages-with-conda_search" class="anchor"></a>Finding Conda packages with <strong>conda_search</strong>
</h2>
<p>If the user is unsure of the exact name, or version of a tool available on conda, they can use the <strong>conda_search</strong> function.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/conda_search.html">conda_search</a></span><span class="op">(</span><span class="st">"salmon"</span>, pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span></code></pre></div>
<pre><code>##      name version                                    channel
## 2  salmon   0.8.2 https://conda.anaconda.org/bioconda/osx-64
## 3  salmon   0.9.0 https://conda.anaconda.org/bioconda/osx-64
## 5  salmon   0.9.1 https://conda.anaconda.org/bioconda/osx-64
## 6  salmon  0.10.0 https://conda.anaconda.org/bioconda/osx-64
## 7  salmon  0.10.1 https://conda.anaconda.org/bioconda/osx-64
## 9  salmon  0.10.2 https://conda.anaconda.org/bioconda/osx-64
## 11 salmon  0.11.3 https://conda.anaconda.org/bioconda/osx-64
## 12 salmon  0.12.0 https://conda.anaconda.org/bioconda/osx-64
## 14 salmon  0.13.0 https://conda.anaconda.org/bioconda/osx-64
## 15 salmon  0.13.1 https://conda.anaconda.org/bioconda/osx-64
## 17 salmon  0.14.0 https://conda.anaconda.org/bioconda/osx-64
## 20 salmon  0.14.1 https://conda.anaconda.org/bioconda/osx-64
## 22 salmon  0.14.2 https://conda.anaconda.org/bioconda/osx-64
## 23 salmon  0.15.0 https://conda.anaconda.org/bioconda/osx-64
## 24 salmon   1.0.0 https://conda.anaconda.org/bioconda/osx-64
## 25 salmon   1.1.0 https://conda.anaconda.org/bioconda/osx-64
## 26 salmon   1.2.0 https://conda.anaconda.org/bioconda/osx-64
## 27 salmon   1.2.1 https://conda.anaconda.org/bioconda/osx-64
## 28 salmon   1.3.0 https://conda.anaconda.org/bioconda/osx-64
## 29 salmon   1.4.0 https://conda.anaconda.org/bioconda/osx-64</code></pre>
<pre><code>## [1] TRUE</code></pre>
<p>Specific package versions can be searched for using the conda format i.e. “salmon==1.3”, “salmon&gt;=1.3” or “salmon&lt;=1.3”. Searches will also find close matches for incorrect queries. Channels to search in can be controlled with <em>channels</em> parameter.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/conda_search.html">conda_search</a></span><span class="op">(</span><span class="st">"salmon&lt;=1.0"</span>, pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span></code></pre></div>
<pre><code>##      name version                                    channel
## 2  salmon   0.8.2 https://conda.anaconda.org/bioconda/osx-64
## 3  salmon   0.9.0 https://conda.anaconda.org/bioconda/osx-64
## 5  salmon   0.9.1 https://conda.anaconda.org/bioconda/osx-64
## 6  salmon  0.10.0 https://conda.anaconda.org/bioconda/osx-64
## 7  salmon  0.10.1 https://conda.anaconda.org/bioconda/osx-64
## 9  salmon  0.10.2 https://conda.anaconda.org/bioconda/osx-64
## 11 salmon  0.11.3 https://conda.anaconda.org/bioconda/osx-64
## 12 salmon  0.12.0 https://conda.anaconda.org/bioconda/osx-64
## 14 salmon  0.13.0 https://conda.anaconda.org/bioconda/osx-64
## 15 salmon  0.13.1 https://conda.anaconda.org/bioconda/osx-64
## 17 salmon  0.14.0 https://conda.anaconda.org/bioconda/osx-64
## 20 salmon  0.14.1 https://conda.anaconda.org/bioconda/osx-64
## 22 salmon  0.14.2 https://conda.anaconda.org/bioconda/osx-64
## 23 salmon  0.15.0 https://conda.anaconda.org/bioconda/osx-64</code></pre>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/conda_search.html">conda_search</a></span><span class="op">(</span><span class="st">"salmo"</span>, pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p><br></p>
</div>
<div id="export-of-conda-environments-to-yaml-files-using-export_condaenv-" class="section level2">
<h2 class="hasAnchor">
<a href="#export-of-conda-environments-to-yaml-files-using-export_condaenv-" class="anchor"></a>Export of Conda environments to YAML files using <strong>export_CondaEnv</strong>.</h2>
<p>The <strong>export_CondaEnv</strong> function allows the user to export the environment information to a <em>.yml</em> file. These environment YAML files contain all essential information about the package, allowing for reproducibility and easy distribution of Conda system configuration for collaboration.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">yml_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"herper_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/format.html">format</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html">Sys.Date</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"%Y%m%d"</span><span class="op">)</span>, <span class="st">".yml"</span><span class="op">)</span>
<span class="fu"><a href="../reference/export_CondaEnv.html">export_CondaEnv</a></span><span class="op">(</span><span class="st">"herper"</span>, <span class="va">yml_name</span>, pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "herper_20210222.yml"</code></pre>
<p><br></p>
<p>The YAML export will contain all packages in the environment by default. If the user wants to only export the packages that were specifically installed and not their dependencies they can use the <em>depends</em> paramter.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">yml_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"herper_nodeps_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/format.html">format</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html">Sys.Date</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"%Y%m%d"</span><span class="op">)</span>, <span class="st">".yml"</span><span class="op">)</span>
<span class="fu"><a href="../reference/export_CondaEnv.html">export_CondaEnv</a></span><span class="op">(</span><span class="st">"herper"</span>, <span class="va">yml_name</span>, depends <span class="op">=</span> <span class="cn">FALSE</span>, pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "herper_nodeps_20210222.yml"</code></pre>
<p><br></p>
</div>
<div id="import-of-conda-environments-from-yaml-files-using-import_condaenv-" class="section level2">
<h2 class="hasAnchor">
<a href="#import-of-conda-environments-from-yaml-files-using-import_condaenv-" class="anchor"></a>Import of Conda environments from YAML files using <strong>import_CondaEnv</strong>.</h2>
<p>The <strong>import_CondaEnv</strong> function allows the user to create a new conda environment from a <em>.yml</em> file. These can be previously exported from <strong>export_CondaEnv</strong>, conda, renv or manually created.</p>
<p>Users can simply provide a path to the YAML file for import. They can also specify the environment name, but by default the name will be taken from the YAML.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">testYML</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/test.yml"</span>,package<span class="op">=</span><span class="st">"Herper"</span><span class="op">)</span>
<span class="fu"><a href="../reference/export_CondaEnv.html">import_CondaEnv</a></span><span class="op">(</span>yml_import<span class="op">=</span><span class="va">testYML</span>, pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span></code></pre></div>
<p><br></p>
</div>
<div id="checking-for-existing-environments-with-list_condaenv" class="section level2">
<h2 class="hasAnchor">
<a href="#checking-for-existing-environments-with-list_condaenv" class="anchor"></a>Checking for existing environments with <strong>list_CondaEnv</strong>
</h2>
<p>The <strong>list_CondaEnv</strong> function allows users to check what environments already exist within the given conda build.</p>
<p>If the User is using multiple builds of conda and wants to check environments across all them, they can include the parameter <em>allCondas = TRUE</em>.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/list_CondaEnv.html">list_CondaEnv</a></span><span class="op">(</span>pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span></code></pre></div>
<pre><code>##                                                                    conda path                 env
## 1 /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/RtmpE46kLy/rr/Test HerperTestPkg_0.1.0
## 2 /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/RtmpE46kLy/rr/Test              herper
## 3 /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/RtmpE46kLy/rr/Test             my_test
## 4 /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/RtmpE46kLy/rr/Test        r-reticulate
## 5 /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/RtmpE46kLy/rr/Test          seqCNA_env</code></pre>
<p><br></p>
</div>
<div id="checking-for-packages-in-an-environment-with-list_condapkgs" class="section level2">
<h2 class="hasAnchor">
<a href="#checking-for-packages-in-an-environment-with-list_condapkgs" class="anchor"></a>Checking for packages in an environment with <strong>list_CondaPkgs</strong>
</h2>
<p>The <strong>list_CondaPkgs</strong> function allows users to check what packages are installed in a given environment.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/list_CondaPkgs.html">list_CondaPkgs</a></span><span class="op">(</span><span class="st">"my_test"</span>, pathToMiniConda <span class="op">=</span> <span class="va">myMiniconda</span><span class="op">)</span></code></pre></div>
<pre><code>##                  name   version     channel platform
## 1            brotlipy     0.7.0 conda-forge   osx-64
## 2     ca-certificates 2020.12.5 conda-forge   osx-64
## 3             certifi 2020.12.5 conda-forge   osx-64
## 4                cffi    1.14.4 conda-forge   osx-64
## 5             chardet     4.0.0 conda-forge   osx-64
## 6               click     7.1.2 conda-forge   noarch
## 7         coloredlogs      15.0 conda-forge   osx-64
## 8           colormath     3.0.0 conda-forge   noarch
## 9        cryptography     3.4.4 conda-forge   osx-64
## 10             cycler    0.10.0 conda-forge   noarch
## 11          decorator     4.4.2 conda-forge   noarch
## 12           freetype    2.10.4 conda-forge   osx-64
## 13             future    0.18.2 conda-forge   osx-64
## 14      humanfriendly       9.1 conda-forge   osx-64
## 15               idna      2.10 conda-forge   noarch
## 16 importlib-metadata     3.4.0 conda-forge   osx-64
## 17             jinja2    2.11.3 conda-forge   noarch
## 18               jpeg        9d conda-forge   osx-64
## 19         kiwisolver     1.3.1 conda-forge   osx-64
## 20              lcms2      2.12 conda-forge   osx-64
## 21            libblas     3.9.0 conda-forge   osx-64
## 22           libcblas     3.9.0 conda-forge   osx-64
## 23             libcxx    11.0.1 conda-forge   osx-64
## 24             libffi     3.2.1    bioconda   osx-64
## 25        libgfortran     5.0.0 conda-forge   osx-64
## 26       libgfortran5     9.3.0 conda-forge   osx-64
## 27          liblapack     3.9.0 conda-forge   osx-64
## 28        libopenblas    0.3.12 conda-forge   osx-64
## 29             libpng    1.6.37 conda-forge   osx-64
## 30            libtiff     4.2.0 conda-forge   osx-64
## 31       libwebp-base     1.2.0 conda-forge   osx-64
## 32        llvm-openmp    11.0.1 conda-forge   osx-64
## 33              lz4-c     1.9.3 conda-forge   osx-64
## 34           lzstring     1.0.4 conda-forge   noarch
## 35           markdown     3.3.3 conda-forge   noarch
## 36         markupsafe     1.1.1 conda-forge   osx-64
## 37    matplotlib-base     3.3.4 conda-forge   osx-64
## 38            multiqc       1.9    bioconda   noarch
## 39            ncurses       6.2 conda-forge   osx-64
## 40           networkx       2.5 conda-forge   noarch
## 41              numpy    1.19.5 conda-forge   osx-64
## 42            olefile      0.46 conda-forge   noarch
## 43            openssl    1.1.1j conda-forge   osx-64
## 44             pillow     8.1.0 conda-forge   osx-64
## 45                pip    21.0.1 conda-forge   noarch
## 46          pycparser      2.20 conda-forge   noarch
## 47          pyopenssl    20.0.1 conda-forge   noarch
## 48          pyparsing     2.4.7 conda-forge   noarch
## 49            pysocks     1.7.1 conda-forge   osx-64
## 50             python    3.6.11 conda-forge   osx-64
## 51    python-dateutil     2.8.1 conda-forge   noarch
## 52         python_abi       3.6 conda-forge   osx-64
## 53             pyyaml     5.4.1 conda-forge   osx-64
## 54           readline       8.0 conda-forge   osx-64
## 55           requests    2.25.1 conda-forge   noarch
## 56         setuptools    49.6.0 conda-forge   osx-64
## 57         simplejson     3.8.1    bioconda   osx-64
## 58                six    1.15.0 conda-forge   noarch
## 59            spectra    0.0.11 conda-forge   noarch
## 60             sqlite    3.34.0 conda-forge   osx-64
## 61                 tk    8.6.10 conda-forge   osx-64
## 62            tornado       6.1 conda-forge   osx-64
## 63  typing_extensions   3.7.4.3 conda-forge   noarch
## 64            urllib3    1.26.3 conda-forge   noarch
## 65              wheel    0.36.2 conda-forge   noarch
## 66                 xz     5.2.5 conda-forge   osx-64
## 67               yaml     0.2.5 conda-forge   osx-64
## 68               zipp     3.4.0 conda-forge   noarch
## 69               zlib    1.2.11 conda-forge   osx-64
## 70               zstd     1.4.8 conda-forge   osx-64</code></pre>
<p><br><br></p>
</div>
<div id="acknowledgements" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgements" class="anchor"></a>Acknowledgements</h2>
<p>Thank you to Ji-Dung Luo and Wei Wang for testing/vignette review/critical feedback and Ziwei Liang for their support.</p>
<p><br></p>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h2>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.4 (2021-02-15)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.7
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] seqCNA_1.36.0       seqCNA.annot_1.26.0 adehabitatLT_0.3.25 CircStats_0.2-6    
##  [5] boot_1.3-26         MASS_7.3-53         adehabitatMA_0.3.14 ade4_1.7-16        
##  [9] sp_1.4-5            doSNOW_1.0.19       snow_0.4-3          iterators_1.0.13   
## [13] foreach_1.5.1       GLAD_2.54.0         Herper_1.1.1        reticulate_1.18    
## [17] BiocStyle_2.18.1   
## 
## loaded via a namespace (and not attached):
##  [1] progress_1.2.2      xfun_0.21           lattice_0.20-41     vctrs_0.3.6        
##  [5] htmltools_0.5.1.1   yaml_2.2.1          rlang_0.4.10        pkgdown_1.6.1      
##  [9] withr_2.4.1         lifecycle_1.0.0     stringr_1.4.0       ragg_1.1.0         
## [13] codetools_0.2-18    memoise_2.0.0       evaluate_0.14       knitr_1.31         
## [17] fastmap_1.1.0       parallel_4.0.4      highr_0.8           Rcpp_1.0.6         
## [21] BiocManager_1.30.10 cachem_1.0.4        desc_1.2.0          jsonlite_1.7.2     
## [25] systemfonts_1.0.1   fs_1.5.0            textshaping_0.3.0   rjson_0.2.20       
## [29] hms_1.0.0           digest_0.6.27       stringi_1.5.3       bookdown_0.21      
## [33] grid_4.0.4          rprojroot_2.0.2     tools_4.0.4         magrittr_2.0.1     
## [37] crayon_1.4.1        pkgconfig_2.0.3     ellipsis_0.3.1      Matrix_1.3-2       
## [41] prettyunits_1.1.1   assertthat_0.2.1    rmarkdown_2.7       R6_2.5.0           
## [45] compiler_4.0.4</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Matt Paul, Thomas Carroll, Doug Barrows.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
